sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConsumer.cs

```diff
@@ -291,9 +292,20 @@ protected EventHubConsumer()
                                                                           [EnumeratorCancellation] CancellationToken cancellationToken = default)
         {
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            EventHubsEventSource.Log.SubscribeToPartitionStart(EventHubName, PartitionId);
 
-            var maximumQueuedEvents = Math.Min((Options.PrefetchCount / 4), (BackgroundPublishReceiveBatchSize * 2));
-            (Guid Identifier, ChannelReader<EventData> ChannelReader) subscription = SubscribeToChannel(EventHubName, PartitionId, ConsumerGroup, maximumQueuedEvents, cancellationToken);
+            (Guid Identifier, ChannelReader<EventData> ChannelReader) subscription;
+
+            try
+            {
+                var maximumQueuedEvents = Math.Min((Options.PrefetchCount / 4), (BackgroundPublishReceiveBatchSize * 2));
+                subscription = SubscribeToChannel(EventHubName, PartitionId, ConsumerGroup, maximumQueuedEvents, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                EventHubsEventSource.Log.SubscribeToPartitionError(EventHubName, PartitionId, ex.Message);
+                throw;
+            }
```

Comment:
Why don't we merge the `try-catch` and `try-finally` blocks into a single `try-catch-finally`? Is the `SubscribeToPartitionError` supposed to be called only on `SubscribeToChannel` failure?

Reply:
We cannot due to the inability to have a `catch` block in the scope where a `yield return` is being performed.   We have a small logging gap in the event that the `EnumerateChannel` throws, though we have the same restriction around `try/catch` in there, as we're using `yield` as well.

I tried to capture the case where possible by chopping up the blocks.   I'm open to trying to close that gap, but I haven't yet figured out a reasonable way to do so.   I think it's more a nice-to-have than critical, since there is no external service call involved; filling the channel uses the standard `Receive` path, so we'll see any issues between the client and service.